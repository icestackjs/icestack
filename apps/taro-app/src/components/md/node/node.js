"use strict";function t(r){"@babel/helpers - typeof";return(t="function"==typeof Symbol&&"symbol"==typeof Symbol.iterator?function(t){return typeof t}:function(t){return t&&"function"==typeof Symbol&&t.constructor===Symbol&&t!==Symbol.prototype?"symbol":typeof t})(r)}function r(t,r,i){return r=e(r),r in t?Object.defineProperty(t,r,{value:i,enumerable:!0,configurable:!0,writable:!0}):t[r]=i,t}function e(r){var e=i(r,"string");return"symbol"===t(e)?e:String(e)}function i(r,e){if("object"!==t(r)||null===r)return r;var i=r[Symbol.toPrimitive];if(void 0!==i){var o=i.call(r,e||"default");if("object"!==t(o))return o;throw new TypeError("@@toPrimitive must return a primitive value.")}return("string"===e?String:Number)(r)}Component({data:{ctrl:{},isiOS:wx.getSystemInfoSync().system.includes("iOS")},properties:{childs:Array,opts:Array},options:{addGlobalClass:!0},attached:function(){this.triggerEvent("add",this,{bubbles:!0,composed:!0})},methods:{noop:function(){},getNode:function(t){try{for(var r=t.split("_"),e=this.data.childs[r[0]],i=1;i<r.length;i++)e=e.children[r[i]];return e}catch(t){return{text:"",attrs:{},children:[]}}},play:function(t){if(this.root.triggerEvent("play"),this.root.data.pauseVideo){for(var r=!1,e=t.target.id,i=this.root._videos.length;i--;)this.root._videos[i].id===e?r=!0:this.root._videos[i].pause();if(!r){var o=wx.createVideoContext(e,this);o.id=e,this.root.playbackRate&&o.playbackRate(this.root.playbackRate),this.root._videos.push(o)}}},imgTap:function(t){var r=this.getNode(t.target.dataset.i);if(r.a)return this.linkTap(r.a);if(!r.attrs.ignore&&(this.root.triggerEvent("imgtap",r.attrs),this.root.data.previewImg)){var e=this.root.imgList[r.i];wx.previewImage({showmenu:this.root.data.showImgMenu,current:e,urls:this.root.imgList})}},imgLoad:function(t){var e,i=t.target.dataset.i,o=this.getNode(i);o.w?(this.data.opts[1]&&!this.data.ctrl[i]||-1===this.data.ctrl[i])&&(e=1):e=t.detail.width,e&&this.setData(r({},"ctrl."+i,e)),this.checkReady()},checkReady:function(){var t=this;this.root.data.lazyLoad||(this.root.imgList._unloadimgs-=1,this.root.imgList._unloadimgs||setTimeout(function(){t.root.getRect().then(function(r){t.root.triggerEvent("ready",r)}).catch(function(){t.root.triggerEvent("ready",{})})},350))},linkTap:function(t){var r=t.currentTarget?this.getNode(t.currentTarget.dataset.i):{},e=r.attrs||t,i=e.href;this.root.triggerEvent("linktap",Object.assign({innerText:this.root.getText(r.children||[])},e)),i&&("#"===i[0]?this.root.navigateTo(i.substring(1)).catch(function(){}):i.split("?")[0].includes("://")?this.root.data.copyLink&&wx.setClipboardData({data:i,success:function(){return wx.showToast({title:"链接已复制"})}}):wx.navigateTo({url:i,fail:function(){wx.switchTab({url:i,fail:function(){}})}}))},mediaError:function(t){var e=t.target.dataset.i,i=this.getNode(e);if("video"===i.name||"audio"===i.name){var o=(this.data.ctrl[e]||0)+1;if(o>i.src.length&&(o=0),o<i.src.length)return this.setData(r({},"ctrl."+e,o))}else"img"===i.name&&(this.data.opts[2]&&this.setData(r({},"ctrl."+e,-1)),this.checkReady());this.root&&this.root.triggerEvent("error",{source:i.name,attrs:i.attrs,errMsg:t.detail.errMsg})}}});