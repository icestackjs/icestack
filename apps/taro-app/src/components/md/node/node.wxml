<wxs module="isInline">var e={abbr:!0,b:!0,big:!0,code:!0,del:!0,em:!0,i:!0,ins:!0,label:!0,q:!0,small:!0,span:!0,strong:!0,sub:!0,sup:!0};module.exports=function(n,i){return e[n]||-1!==(i||"").indexOf("inline")};</wxs><template name="el"><block wx:if="{{n.name==='img'}}"><rich-text wx:if="{{n.t}}" style="display:{{n.t}}" nodes="<img class='_img' style='{{n.attrs.style}}' src='{{n.attrs.src}}'>" data-i="{{i}}" catchtap="imgTap"/><block wx:else><image wx:if="{{(opts[1]&&!ctrl[i])||ctrl[i]<0}}" class="_img" style="{{n.attrs.style}}" src="{{ctrl[i]<0?opts[2]:opts[1]}}" mode="widthFix"/><image id="{{n.attrs.id||('n'+i)}}" class="_img {{n.attrs.class}}" style="{{ctrl['e'+i]?'border:1px dashed black;padding:3px;':''}}{{ctrl[i]===-1?'display:none;':''}}width:{{ctrl[i]||1}}px;height:{{ctrl['h'+i]||1}}px;{{n.attrs.style}}" src="{{n.attrs.src}}" mode="{{!n.h?'widthFix':(!n.w?'heightFix':'scaleToFill')}}" lazy-load="{{opts[0]}}" webp="{{n.webp}}" show-menu-by-longpress="{{opts[3]&&!n.attrs.ignore}}" data-i="{{i}}" bindload="imgLoad" binderror="mediaError" catchtap="imgTap" bindlongpress="noop"/></block></block><block wx:elif="{{n.type==='text'}}"><text wx:if="{{!ctrl['e'+i]}}" data-i="{{i}}" user-select="{{opts[4]}}" decode="{{!opts[5]}}" bindtap="editStart">{{n.text}}<text wx:if="{{!n.text}}" style="color:gray">{{opts[6]||'请输入'}}</text></text><text wx:elif="{{ctrl['e'+i]===1}}" data-i="{{i}}" style="border:1px dashed black;min-width:50px;width:auto;padding:5px;display:block" catchtap="editStart">{{n.text}}<text wx:if="{{!n.text}}" style="color:gray">{{opts[6]||'请输入'}}</text></text><textarea wx:else style="border:1px dashed black;min-width:50px;width:auto;padding:5px" auto-height maxlength="-1" focus="{{ctrl['e'+i]===3}}" value="{{n.text}}" data-i="{{i}}" bindinput="editInput" bindblur="editEnd"/></block><text wx:elif="{{n.name==='br'}}">\n</text><view wx:elif="{{n.name==='a'}}" id="{{n.attrs.id}}" class="{{n.attrs.href?'_a ':''}}{{n.attrs.class}}" hover-class="_hover" style="display:inline;{{n.attrs.style}}" data-i="{{i}}" catchtap="linkTap"><node childs="{{n.children}}" opts="{{[opts[0],opts[1],opts[2],opts[3],opts[4],opts[5],opts[6],opts[7]+i+'_']}}" style="display:inherit"/></view><video bindtap="mediaTap" wx:elif="{{n.name==='video'}}" id="{{n.attrs.id}}" class="{{n.attrs.class}}" style="{{n.attrs.style}}" autoplay="{{n.attrs.autoplay}}" controls="{{n.attrs.controls}}" loop="{{n.attrs.loop}}" muted="{{n.attrs.muted}}" object-fit="{{n.attrs['object-fit']}}" poster="{{n.attrs.poster}}" src="{{n.src[ctrl[i]||0]}}" data-i="{{i}}" bindplay="play" binderror="mediaError"/><my-audio bindtap="mediaTap" wx:elif="{{n.name=='audio'}}" id="{{n.attrs.id}}" class="{{n.attrs.class}}" style="{{n.attrs.style}}" author="{{n.attrs.author}}" controls="{{n.attrs.controls}}" autoplay="{{n.attrs.autoplay}}" loop="{{n.attrs.loop}}" name="{{n.attrs.name}}" poster="{{n.attrs.poster}}" src="{{n.src[ctrl[i]||0]}}" data-i="{{i}}" data-source="audio" bindplay="play" binderror="mediaError"/><card wx:elif="{{n.name=='card'}}" class="{{n.attrs.class}}" style="{{n.attrs.style}}"/><rich-text wx:else id="{{n.attrs.id}}" style="{{n.f}}" user-select="{{opts[4]}}" nodes="{{[n]}}"/></template><block wx:for="{{childs}}" wx:for-item="n1" wx:for-index="i1" wx:key="i1"><template wx:if="{{(opts[5]?true:!n1.c)&&(!n1.children||n1.name==='a'||!(opts[5]?true:isInline(n1.name,n1.attrs.style)))}}" is="el" data="{{n:n1,i:''+i1,opts:opts,ctrl:ctrl}}"/><view wx:else data-i="{{''+i1}}" bindtap="nodeTap" id="{{n1.attrs.id}}" class="_{{n1.name}} {{n1.attrs.class}}" style="{{ctrl['e'+i1]?'border:1px solid black;padding:5px;display:block;':''}}{{n1.attrs.style}}"><block wx:for="{{n1.children}}" wx:for-item="n2" wx:for-index="i2" wx:key="i2"><template wx:if="{{(opts[5]?true:!n2.c)&&(!n2.children||n2.name==='a'||!(opts[5]?true:isInline(n2.name,n2.attrs.style)))}}" is="el" data="{{n:n2,i:i1+'_'+i2,opts:opts,ctrl:ctrl}}"/><view wx:else data-i="{{i1+'_'+i2}}" bindtap="nodeTap" id="{{n2.attrs.id}}" class="_{{n2.name}} {{n2.attrs.class}}" style="{{ctrl['e'+i1+'_'+i2]?'border:1px solid black;padding:5px;display:block;':''}}{{n2.attrs.style}}"><block wx:for="{{n2.children}}" wx:for-item="n3" wx:for-index="i3" wx:key="i3"><template wx:if="{{(opts[5]?true:!n3.c)&&(!n3.children||n3.name==='a'||!(opts[5]?true:isInline(n3.name,n3.attrs.style)))}}" is="el" data="{{n:n3,i:i1+'_'+i2+'_'+i3,opts:opts,ctrl:ctrl}}"/><view wx:else data-i="{{i1+'_'+i2+'_'+i3}}" bindtap="nodeTap" id="{{n3.attrs.id}}" class="_{{n3.name}} {{n3.attrs.class}}" style="{{ctrl['e'+i1+'_'+i2+'_'+i3]?'border:1px solid black;padding:5px;display:block;':''}}{{n3.attrs.style}}"><block wx:for="{{n3.children}}" wx:for-item="n4" wx:for-index="i4" wx:key="i4"><template wx:if="{{(opts[5]?true:!n4.c)&&(!n4.children||n4.name==='a'||!(opts[5]?true:isInline(n4.name,n4.attrs.style)))}}" is="el" data="{{n:n4,i:i1+'_'+i2+'_'+i3+'_'+i4,opts:opts,ctrl:ctrl}}"/><view wx:else data-i="{{i1+'_'+i2+'_'+i3+'_'+i4}}" bindtap="nodeTap" id="{{n4.attrs.id}}" class="_{{n4.name}} {{n4.attrs.class}}" style="{{ctrl['e'+i1+'_'+i2+'_'+i3+'_'+i4]?'border:1px solid black;padding:5px;display:block;':''}}{{n4.attrs.style}}"><block wx:for="{{n4.children}}" wx:for-item="n5" wx:for-index="i5" wx:key="i5"><template wx:if="{{(opts[5]?true:!n5.c)&&(!n5.children||n5.name==='a'||!(opts[5]?true:isInline(n5.name,n5.attrs.style)))}}" is="el" data="{{n:n5,i:i1+'_'+i2+'_'+i3+'_'+i4+'_'+i5,opts:opts,ctrl:ctrl}}"/><node wx:else id="{{n5.attrs.id}}" class="_{{n5.name}} {{n5.attrs.class}}" style="{{n5.attrs.style}}" childs="{{n5.children}}" opts="{{[opts[0],opts[1],opts[2],opts[3],opts[4],opts[5],opts[6],opts[7]+i1+'_'+i2+'_'+i3+'_'+i4+'_'+i5+'_']}}"/></block></view></block></view></block></view></block></view></block>