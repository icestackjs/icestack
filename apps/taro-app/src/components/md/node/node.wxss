.hl-code,.hl-pre{color:#ccc;background:0 0;font-family:Consolas,Monaco,'Andale Mono','Ubuntu Mono',monospace;font-size:1em;text-align:left;white-space:pre;word-spacing:normal;word-break:normal;word-wrap:normal;line-height:1.5;-moz-tab-size:4;-o-tab-size:4;tab-size:4;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;hyphens:none}.hl-pre{padding:1em;margin:.5em 0;overflow:auto}.hl-pre{background:#2d2d2d}.hl-block-comment,.hl-cdata,.hl-comment,.hl-doctype,.hl-prolog{color:#999}.hl-punctuation{color:#ccc}.hl-attr-name,.hl-deleted,.hl-namespace,.hl-tag{color:#e2777a}.hl-function-name{color:#6196cc}.hl-boolean,.hl-function,.hl-number{color:#f08d49}.hl-class-name,.hl-constant,.hl-property,.hl-symbol{color:#f8c555}.hl-atrule,.hl-builtin,.hl-important,.hl-keyword,.hl-selector{color:#cc99cd}.hl-attr-value,.hl-char,.hl-regex,.hl-string,.hl-variable{color:#7ec699}.hl-entity,.hl-operator,.hl-url{color:#67cdcc}.hl-bold,.hl-important{font-weight:700}.hl-italic{font-style:italic}.hl-entity{cursor:help}.hl-inserted{color:green}.hl-pre[data-line]{position:relative;padding:1em 0 1em 3em}.line-highlight{position:absolute;left:0;right:0;padding:inherit 0;margin-top:1em;background:hsla(24,20%,50%,.08);background:linear-gradient(to right,hsla(24,20%,50%,.1) 70%,hsla(24,20%,50%,0));pointer-events:none;line-height:inherit;white-space:pre}@media print{.line-highlight{-webkit-print-color-adjust:exact;color-adjust:exact}}.line-highlight:before,.line-highlight[data-end]:after{content:attr(data-start);position:absolute;top:.4em;left:.6em;min-width:1em;padding:0 .5em;background-color:hsla(24,20%,50%,.4);color:#f4f1ef;font:bold 65%/1.5 sans-serif;text-align:center;vertical-align:.3em;border-radius:999px;text-shadow:none;box-shadow:0 1px #fff}.line-highlight[data-end]:after{content:attr(data-end);top:auto;bottom:.4em}.line-numbers .line-highlight:after,.line-numbers .line-highlight:before{content:none}.hl-pre[id].linkable-line-numbers span.line-numbers-rows{pointer-events:all}.hl-pre[id].linkable-line-numbers span.line-numbers-rows>span:before{cursor:pointer}.hl-pre[id].linkable-line-numbers span.line-numbers-rows>span:hover:before{background-color:rgba(128,128,128,.2)}.hl-pre.line-numbers{position:relative;padding-left:3.8em;counter-reset:linenumber}.hl-pre.line-numbers>.hl-code{position:relative;white-space:inherit}.line-numbers .line-numbers-rows{position:absolute;pointer-events:none;top:0;font-size:100%;left:-3.8em;width:3em;letter-spacing:-1px;border-right:1px solid #999;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.line-numbers-rows>span{display:block;counter-increment:linenumber}.line-numbers-rows>span:before{content:counter(linenumber);color:#999;display:block;padding-right:.8em;text-align:right}div..hl-code-toolbar{position:relative}div..hl-code-toolbar>.toolbar{position:absolute;z-index:10;top:.3em;right:.2em;transition:opacity .3s ease-in-out;opacity:0}div..hl-code-toolbar:hover>.toolbar{opacity:1}div..hl-code-toolbar:focus-within>.toolbar{opacity:1}div..hl-code-toolbar>.toolbar>.toolbar-item{display:inline-block}div..hl-code-toolbar>.toolbar>.toolbar-item>a{cursor:pointer}div..hl-code-toolbar>.toolbar>.toolbar-item>button{background:0 0;border:0;color:inherit;font:inherit;line-height:normal;overflow:visible;padding:0;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none}div..hl-code-toolbar>.toolbar>.toolbar-item>a,div..hl-code-toolbar>.toolbar>.toolbar-item>button,div..hl-code-toolbar>.toolbar>.toolbar-item>span{color:#bbb;font-size:.8em;padding:0 .5em;background:#f5f2f0;background:rgba(224,224,224,.2);box-shadow:0 2px 0 0 rgba(0,0,0,.2);border-radius:.5em}div..hl-code-toolbar>.toolbar>.toolbar-item>a:focus,div..hl-code-toolbar>.toolbar>.toolbar-item>a:hover,div..hl-code-toolbar>.toolbar>.toolbar-item>button:focus,div..hl-code-toolbar>.toolbar>.toolbar-item>button:hover,div..hl-code-toolbar>.toolbar>.toolbar-item>span:focus,div..hl-code-toolbar>.toolbar>.toolbar-item>span:hover{color:inherit;text-decoration:none}span.inline-color-wrapper{background:url(data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyIDIiPjxwYXRoIGZpbGw9ImdyYXkiIGQ9Ik0wIDBoMnYySDB6Ii8+PHBhdGggZmlsbD0id2hpdGUiIGQ9Ik0wIDBoMXYxSDB6TTEgMWgxdjFIMXoiLz48L3N2Zz4=);background-position:center;background-size:110%;display:inline-block;height:1.333ch;width:1.333ch;margin:0 .333ch;box-sizing:border-box;border:1px solid #fff;outline:1px solid rgba(0,0,0,.5);overflow:hidden}span.inline-color{display:block;height:120%;width:120%}.md-p{margin-block-start:1em;margin-block-end:1em}.md-blockquote,.md-table{margin-bottom:16px}.md-table{box-sizing:border-box;width:100%;overflow:auto;border-spacing:0;border-collapse:collapse}.md-tr{background-color:#fff;border-top:1px solid #c6cbd1}.md-table .md-tr:nth-child(2n){background-color:#f6f8fa}.md-td,.md-th{padding:6px 13px!important;border:1px solid #dfe2e5}.md-th{font-weight:600}.md-blockquote{padding:0 1em;color:#6a737d;border-left:.25em solid #dfe2e5}.md-code{padding:.2em .4em;font-family:SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace;font-size:85%;background-color:rgba(27,31,35,.05);border-radius:3px}.md-pre .md-code{padding:0;font-size:100%;background:0 0;border:0}._a{padding:1.5px 0 1.5px 0;color:#366092;word-break:break-all}._hover{text-decoration:underline;opacity:.7}._img{max-width:100%;-webkit-touch-callout:none}._b,._strong{font-weight:700}._code{font-family:monospace}._del{text-decoration:line-through}._em,._i{font-style:italic}._h1{font-size:2em}._h2{font-size:1.5em}._h3{font-size:1.17em}._h5{font-size:.83em}._h6{font-size:.67em}._h1,._h2,._h3,._h4,._h5,._h6{display:block;font-weight:700}._ins{text-decoration:underline}._li{display:list-item}._ol{list-style-type:decimal}._ol,._ul{display:block;padding-left:40px;margin:1em 0}._q::before{content:'"'}._q::after{content:'"'}._sub{font-size:smaller;vertical-align:sub}._sup{font-size:smaller;vertical-align:super}._tbody,._tfoot,._thead{display:table-row-group}._tr{display:table-row}._td,._th{display:table-cell;vertical-align:middle}._th{font-weight:700;text-align:center}._ul{list-style-type:disc}._ul ._ul{margin:0;list-style-type:circle}._ul ._ul ._ul{list-style-type:square}._abbr,._b,._code,._del,._em,._i,._ins,._label,._q,._span,._strong,._sub,._sup{display:inline}._blockquote,._div,._p{display:block}