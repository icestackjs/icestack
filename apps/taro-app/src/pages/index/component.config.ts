export default definePageConfig({
  navigationBarTitleText: '组件',
  navigationStyle: 'custom',
})
