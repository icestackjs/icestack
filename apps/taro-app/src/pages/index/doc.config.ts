export default definePageConfig({
  navigationBarTitleText: '文档',
  navigationStyle: 'custom',
})
