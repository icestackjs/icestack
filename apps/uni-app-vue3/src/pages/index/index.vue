<template>
  <view class="content">
    <WeappTailwindcss></WeappTailwindcss>

    <view class="text-gray-900/50 mb-2 before:content-['当前系统主题:']">
      {{ themeRef }}
    </view>
    <view class="space-y-[20rpx] flex flex-col items-center">
      <view
        class="bg-[#010101] h-16 w-16 rounded-[20rpx] text-white flex justify-center items-center after:content-['hover']"
        hover-class="bg-[gray] after:!content-['good!']"></view>
      <view
        class="grid grid-cols-3 divide-x-[10px] divide-[#010101] divide-solid">
        <div :class="classArray">1</div>
        <div>2</div>
        <div :class="classArray">3</div>
      </view>
      <view
        class="w-32 py-2 rounded-md font-semibold text-white bg-pink-500 ring-4 ring-pink-300">
        Default
      </view>
      <view>
        <button class="text-[#fff]" :class="buttonClass" @click="increment">
          click here to inc {{ count }}
        </button>
      </view>

      <view class="test">@apply</view>
    </view>
  </view>
</template>

<script setup lang="ts">
import WeappTailwindcss from '@/components/WeappTailwindcss.vue'
import { useCounterStore } from '@/stores/counter'

const store = useCounterStore()
const { count } = storeToRefs(store)
const { increment } = store
const buttonColors = [
  'bg-[#000]',
  'bg-[#111]',
  'bg-[#222]',
  'bg-[#333]',
  'bg-[#444]',
  'bg-[#555]',
  'bg-[#666]',
  'bg-[#777]',
  'bg-[#888]',
  'bg-[#999]',
  'bg-[#aaa]',
  'bg-[#bbb]',
  'bg-[#ccc]',
  'bg-[#ddd]',
  'bg-[#eee]',
  'bg-[#fff]'
]
const title = ref('Hello')
const themeRef = ref(uni.getSystemInfoSync().theme)
const classArray = computed(() => [
  title.value ? 'bg-[#ff00ff]' : undefined,
  {
    'text-[#00ffff]': Boolean(title),
    "bg-[url('https://xxx.com/xx.webp')]": true,
    "bg-[url('https://yyyy.com/ccc.webp')]": true
  }
])
const buttonClass = computed(() => {
  return buttonColors[count.value % buttonColors.length]
})
// #ifdef MP
uni.onThemeChange(({ theme }: { theme: 'dark' | 'light' }) => {
  themeRef.value = theme
})
// #endif
onBeforeUnmount(() => {
  // #ifdef MP
  uni.offThemeChange(() => {
    console.log('offThemeChange')
  })
  // #endif
})

onLoad(() => {
  console.log('欢迎使用uni-app-vite-vue3-tailwindcss模板')
})
</script>

<style lang="scss" scoped>
.logo {
  @apply h-[100rpx] w-[100rpx];
}

.content {
  @apply flex flex-col items-center;
}

.test {
  @apply flex items-center justify-center h-[100px] w-[100px] rounded-[40px] bg-[#123456] bg-opacity-[0.54] text-[#ffffff] #{!important};
}
</style>
