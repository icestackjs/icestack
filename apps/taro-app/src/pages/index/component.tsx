import { View, Text, Button, ViewProps } from '@tarojs/components'
// import { useLoad } from '@tarojs/taro'
import React, { useState, useMemo } from 'react'
import './index.scss'
import { upperFirst } from 'lodash-es'
import ThemeProvider from '@/components/ThemeProvider'
import Taro, { useLoad } from '@tarojs/taro'

const map = {
  button: () => {
    return (
      <View>
        <View className='ice-subtitle'>按钮类型</View>
      </View>
    )
  },
  alert: () => {
    return (
      <View>
        <View className='ice-alert'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-primary'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-success'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
        <View className='ice-alert ice-alert-warning'>
          <Text className='i-mdi-information-outline w-6 h-6'></Text>
          <Text>默认alert</Text>
        </View>
      </View>
    )
  }
}

export default function Index() {
  const [mode, setMode] = useState<'light' | 'dark'>('light')
  const [com, setCom] = useState('')
  useLoad<{ id: string }>((params) => {
    if (params.id) {
      Taro.setNavigationBarTitle({
        title: upperFirst(params.id)
      })
      setCom(params.id)
    }
  })
  const TargetCom = useMemo(() => {
    return map[com]
  }, [com])
  return (
    <ThemeProvider mode={mode}>
      <View className='px-4'>{TargetCom && <TargetCom></TargetCom>}</View>
    </ThemeProvider>
  )
}
