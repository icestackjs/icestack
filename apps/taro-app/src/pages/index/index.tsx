import { View, Text } from '@tarojs/components'
import Taro from '@tarojs/taro'
import { upperFirst } from 'lodash-es'
import { ReactNode, useState } from 'react'
import './index.scss'
import ThemeProvider from '@/components/ThemeProvider'
import { group, i18n } from '../../group'
import ThemeButton from '@/components/ThemeButton'
import Navbar from '@/components/Navbar'
import Footer from '@/components/Footer'
import HomeTitle from '@/components/HomeTitle'

const aliasMap = {
  range: 'Slider',
  toggle: 'Switch'
}

function getAlias(componentName: string) {
  const x = aliasMap[componentName]
  if (x) {
    return '/' + x
  }
  return ''
}

export default function Index() {
  return (
    <ThemeProvider>
      <Navbar>
        <View
          className='text-center h-full flex pl-4 items-center text-lg'
          onClick={() => {
            Taro.setClipboardData({
              data: 'https://github.com/sonofmagic/icestack'
            })
          }}
        >
          <View className='rounded-md border border-solid border-[rgba(31,35,40,0.15)] bg-[rgb(246,248,250)] dark:border-[rgba(205,217,229,0.1)] dark:bg-[rgb(55,62,71)] flex items-center px-2 py-1'>
            <View className='i-mdi-github'></View>
            <View className='text-slate-900 dark:text-white mx-0.5 i-mdi-add'></View>
            <View className='i-mdi-star text-[#daaa3f]'></View>
          </View>
        </View>
      </Navbar>
      <View className='min-h-screen px-5 pt-2 pb-3'>
        <View className='px-4'>
          <HomeTitle />
          <View className='text-slate-600 dark:text-slate-400 text-sm mb-1 text-center'>灵活自由的开源CSS Component生成器</View>

          {/* <md markdown content={'```js\nconst y = \'1111\'\n```\n'}></md> */}
          {/* <View className='text-slate-500 dark:text-slate-300 text-xs mb-1'>tips: 本小程序由于大量使用了 grid 布局和 css 新特性，可能会出现部分机型不兼容的情况</View> */}
          {/* <View
            className='mb-4 text-xs text-slate-500 dark:text-slate-300'
            onClick={() => {
              Taro.setClipboardData({
                data: 'https://github.com/sonofmagic/icestack/tree/main/apps/taro-app'
              })
            }}
          >
            源代码地址:
            <View className='link link-primary'>sonofmagic/icestack/apps/taro-app</View>
          </View> */}
        </View>

        {Object.entries(group).reduce<ReactNode[]>((acc, [groupName, componentNames]) => {
          if (componentNames.length) {
            acc.push(<View className='text-slate-600 dark:text-slate-400 text-sm ml-4 mt-6 mb-2'>{i18n[groupName]}</View>)
            // space-y-3
            const res: ReactNode[] = []
            for (const componentName of componentNames) {
              if (['select', 'diff','tooltip'].includes(componentName)) {
                continue
              }
              res.push(
                <View
                  className='bg-gray-100 dark:bg-sky-300/[0.15] py-2.5 rounded-full pl-5 pr-3.5 flex justify-between items-center'
                  hoverClass='bg-gray-300/50 dark:bg-sky-500/50'
                  key={componentName}
                  onClick={() => {
                    Taro.navigateTo({
                      url: 'component?id=' + componentName
                    })
                  }}
                >
                  <View className='text-sm text-slate-700 dark:text-slate-400'>
                    {upperFirst(componentName)}
                    {getAlias(componentName)} {i18n[componentName]}
                  </View>
                  <View className='i-mdi-chevron-right text-slate-700 dark:text-slate-400'></View>
                </View>
              )
            }
            acc.push(<View className='space-y-3'>{res}</View>)
          }

          return acc
        }, [])}
      </View>
      <ThemeButton></ThemeButton>
      <Footer></Footer>
    </ThemeProvider>
  )
}
