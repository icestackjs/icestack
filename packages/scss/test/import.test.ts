import path from 'node:path'
import { pathToFileURL } from 'node:url'
// import { parse } from '@icestack/postcss-utils'
import { compileScssString } from '@/index'

describe('import', () => {
  it('import ', () => {
    const scss = `@import 'foundation/code', 'foundation/lists';`
    // const ast = parse(scss)
    // console.log(ast)
    const { css } = compileScssString(scss, {
      importers: [
        {
          findFileUrl(url, ctx) {
            // console.log(ctx)
            const ppp = path.resolve(__dirname, url)
            return new URL(pathToFileURL(ppp))
            // if (url === 'foundation/code') {

            // } else if (url === 'foundation/lists') {
            //   return new URL(url.slice(1), pathToFileURL('node_modules'))
            // }
          },
        },
      ],
    })
    expect(css).toMatchSnapshot()
  })
})
