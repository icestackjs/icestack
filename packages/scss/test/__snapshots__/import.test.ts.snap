// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`import > import  1`] = `
"code {
  padding: 0.25em;
  line-height: 0;
}

ul,
ol {
  text-align: left;
}
ul ul,
ul ol,
ol ul,
ol ol {
  padding-bottom: 0;
  padding-left: 0;
}"
`;
